data/transactions/logic/assembler.go

```diff
@@ -775,10 +775,14 @@ func typecheck(expected, got StackType) bool {
 }
 
 func filterFieldsForLineComment(fields []string) []string {
+	prevField := ""
```

Comment:
I think a more robust approach is
1. filter out strings starting with // as comments
2. use field[0] as opcode (in Assemble)
3. knowing amount or arguments per opcode treat rest of the line as a comment

Reply:
That would change the language specification, which I'd rather avoid doing in the context of "let's fix a bug".